app/forms/reset_password_form.rb

```diff
@@ -47,6 +47,8 @@ def handle_valid_password
   end
 
   def update_user
+    @password_matches_existing = user.valid_password?(password)
```

Comment:
This is a relatively expensive comparison, I'm not sure if we'd want to do this on every password reset going forward. It may even be desirable to feature flag it?

Reply:
Looks like we do roughly 100k password resets a day. If we are concerned we could capture this for a subset (5%? 10%?) and still get a meaningful result?

Reply:
Yeah I could set this up as an A/B test to limit the performance impact.

Reply:
Although, in that hypothetical future, I suppose the cost of checking whether the password has changed would be offset slightly by avoiding hashing the new password when it's unchanged.